### PR TITLE
Support positional matching

### PR DESCRIPTION
**Pull request by AstraLuma**
New for 3.10.

### CODE REVIEW

**Comment by AstraLuma** [comment]
Not sure how to handle testing for this, given that the preferred tests won't even compile on <3.10.

**Comment by nicoonoclaste** [comment]
@AstraLuma I'd suggest having a `test_py3.10.py` or such, and [selectively ignoring it in `conftest.py`](https://docs.pytest.org/en/6.2.x/example/pythoncollection.html#customizing-test-collection) based on Python's version number.

**Comment by AstraLuma** [comment]
ok, now i officially have no idea why pytest isn't collecting properly.

**Comment by AstraLuma** [comment]
@nbraud @pathunstrom I think this is ready for review.

**Review by pathunstrom — APPROVED** [review]
I like the methodology for testing forward looking features when they don't break syntax.

**Review by nicoonoclaste — APPROVED** [review]
LGTM, except it broke `lint.sh` on Python versions older than 3.10 (not unlikely on contributors' dev environment)

I wrote a fix for that, please review it and if it's fine by you then `bors r=nbraud,pathunstrom` ?

File: ppb_vector/__init__.py [hunk 1]
```diff
@@ -64,6 +64,8 @@ class Vector:
     x: float
     y: float
 
+    __match_args__ = ('x', 'y')
```
Comment (nicoonoclaste) [inline ppb_vector/__init__.py:67]: Might be nice to add a comment pointing to the right documentation (in the PEP or otherwise) for that dunder ^^

**Comment by AstraLuma** [comment]
Oh, I just fixed it by updating the containers. But I guess that helps with development environments.

**Comment by AstraLuma** [comment]
bors r=nbraud,pathunstrom

**Pull request merged**